pkg/currency/module.go

```diff
@@ -11,10 +12,12 @@ import (
 )
 
 type Module struct {
-	kernel.BackgroundModule
+	kernel.HealthCheckedModule
+	kernel.EssentialBackgroundModule
 	kernel.ServiceStage
-	updaterService UpdaterService
 	logger         log.Logger
+	updaterService UpdaterService
+	healthy        *atomic.Bool
```

Comment:
As the module is only used with a pointer to it, we can directly embed the bool and don't need a pointer indirection